lib/iris/_lazy_data.py

```diff
@@ -135,3 +135,67 @@ def multidim_lazy_stack(stack):
         result = da.stack([multidim_lazy_stack(subarray)
                            for subarray in stack])
     return result
+
+
+def convert_nans_array(array, nans_replacement=None, result_dtype=None):
+    """
+    Convert a :class:`~numpy.ndarray` that may contain one or more NaN values
+    to either a :class:`~numpy.ma.core.MaskedArray` or a
+    :class:`~numpy.ndarray` with the NaN values filled.
+
+    Args:
+
+    * array:
+        The :class:`~numpy.ndarray` to be converted.
+
+    Kwargs:
+
+    * nans_replacement:
+        If `nans_replacement` is None, then raise an exception if the `array`
+        contains any NaN values (default behaviour).
+        If `nans_replacement` is `numpy.ma.masked`, then convert the `array`
+        to a :class:`~numpy.ma.core.MaskedArray`.
```

Comment:
NOTE: something missing here.
We should explain that in 'masked array' mode the input array is copied, but in "fill or no-nans modes", the result is the input array, which may be ***modified in-place***.

> may be modified in-place

P.S. *except* when we do a change of dtype.
A bit slippery that, isn't it :grimacing: ??
Perhaps should just say "in some cases, the input is modified in-place" + leave it at that ?